content/docs/command-reference/exp/remove.md

```diff
@@ -14,9 +14,22 @@ positional arguments:
 
 ## Description
 
-Deletes one or more experiments indicated by name or hash (see `dvc exp run`).
+Deletes one or more experiments, indicated by name (see `dvc exp run`).
 
-With `--queue`, the queue of experiments is cleared.
+With `--queue`, the list of experiments awaiting execution is cleared instead.
+
+> Note that all the checkpoints in an experiment are deleted by this command.
+
+<details>
+
+### What happens to removed experiments?
+
+Internally, `dvc exp` uses actual Git commits under custom Git references (found
+in `.git/refs/exps`). `dvc remove` deletes the references only, rendering the
+commits unreachable. `git` should garbage collect them automatically
+[after some time](https://git-scm.com/docs/git-gc#Documentation/git-gc.txt---pruneltdategt).
```

Comment:
> Continues https://github.com/iterative/dvc.org/pull/2300#pullrequestreview-617332002

This expandable section is in line with the **How does DVC track experiments?** and **How are checkpoints captured by DVC?** sections already published in [`exp run`](https://dvc.org/doc/command-reference/exp/run).

The reason I decided to include them is that `git`-savvy users may want to manipulate experiments manually or even build integrations (and this is probably hard to figure out from the codebase). Note their non-intrusive design:

![image](https://user-images.githubusercontent.com/1477535/112095837-da45de80-8b62-11eb-9ebc-282cdefd6f6d.png)

Reply:
From @pmrowla in https://github.com/iterative/dvc.org/pull/2300#discussion_r600096210:

> I think it is fine to explain that "experiments are stored as custom git references, and are comparable to branch heads, removing an experiment does the same thing as removing a git branch" and stop there.
> I liken this to the github documentation for [deleting a branch](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-branches#deleting-a-branch)... Nowhere in their docs does github mention that they create their own duplicate custom refs.
> A "git savvy" user can use the CLI to inspect all of the refs in the repo and figure out what github is doing

Compelling. Not sure about the exact analogy with GH but still a valid argument. OK, I'll change this PR's scope to simplify the existing implementation detail notes (in `exp run`) instead, and see how to incorporate just a min. of detail here ⌛ 

> Mentioning this kind of things in the documentation is just going to invite user questions along the lines of "why doesn't DVC manually delete commit objects"

Inviting user questions may be a good thing though.